src/socketTx.ts

```diff
@@ -60,15 +60,40 @@ export class SocketTx {
     return allowanceValue.lt(minimumApprovalAmount);
   }
 
-  private _validateSend(send: {
+  private async _validateSend(send: {
     data?: string | undefined;
     to?: string | undefined;
     from?: string | undefined;
   }) {
     if (this.userTxType === UserTxType.FUND_MOVR) {
-      const addresses = Object.values(socketAddresses[this.chainId]);
-      if (!addresses.includes(send.to)) {
-        throw new Error(`${send.to} is not a recognised socket address on chain ${this.chainId}`);
+      // Fetcg route.
+      const routeResponse = await Routes.getActiveRoute({ activeRouteId: this.activeRouteId });
+      if (!routeResponse.success) throw new Error("Error while fetching route.");
+      const route = routeResponse.result;
+
+      // Current User Tx - Bridge Name
+      const currentUserTx = route.userTxs[this.userTxIndex];
+      const bridgeStep = currentUserTx.steps.find((step) => step.type === 'bridge');
+      if (bridgeStep) {
+        const { name: protocolName } = bridgeStep.protocol;
+        if (
+          protocolName === socketConstants.bridges.PolygonBridge &&
+          this.chainId === socketConstants.chains.POLYGON_CHAIN_ID
+        ) {
+          const { address: fromAssetAddress } = bridgeStep.fromAsset;
+          if (fromAssetAddress.toLowerCase() !== send.to?.toLowerCase()) {
+            throw new Error(
+              `Polygon Native Bridge burns the token when withdrawn to Ethereum. ${send.to} does not match the token being burnt.`
+            );
+          }
+        } else {
+          const addresses = Object.values(socketAddresses[this.chainId]);
```

Comment:
This would be executed before even if the it was not `bridgeStep`
Is that intended? 
Now if it is not `bridgeStep`, there will be no check against socket addresses

Reply:
So, the step "bridge" will always be there in the FUND_MOVR UserTx.  
Since you are checking it for the FUND_MOVR tx type, the bridge step will alway be present.